packages/twenty-server/src/engine/core-modules/domain-manager/crons/jobs/check-custom-domain-valid-records.cron.job.ts

```diff
@@ -0,0 +1,56 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { IsNull, Not, Repository } from 'typeorm';
+import chunk from 'lodash.chunk';
+import { WorkspaceActivationStatus } from 'twenty-shared/workspace';
+
+import { Process } from 'src/engine/core-modules/message-queue/decorators/process.decorator';
+import { Processor } from 'src/engine/core-modules/message-queue/decorators/processor.decorator';
+import { MessageQueue } from 'src/engine/core-modules/message-queue/message-queue.constants';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { CustomDomainService } from 'src/engine/core-modules/domain-manager/services/custom-domain.service';
+import { SentryCronMonitor } from 'src/engine/core-modules/cron/sentry-cron-monitor.decorator';
+
+export const CHECK_CUSTOM_DOMAIN_VALID_RECORDS_CRON_PATTERN = '0 * * * *';
+
+@Processor(MessageQueue.cronQueue)
+@SentryCronMonitor(
+  CheckCustomDomainValidRecordsCronJob.name,
+  CHECK_CUSTOM_DOMAIN_VALID_RECORDS_CRON_PATTERN,
+)
+export class CheckCustomDomainValidRecordsCronJob {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    private readonly customDomainService: CustomDomainService,
+  ) {}
+
+  @Process(CheckCustomDomainValidRecordsCronJob.name)
+  async handle(): Promise<void> {
+    const workspaces = await this.workspaceRepository.find({
+      where: {
+        activationStatus: WorkspaceActivationStatus.ACTIVE,
+        customDomain: Not(IsNull()),
+      },
+      select: ['id', 'customDomain', 'isCustomDomainEnabled'],
+    });
+
+    const chunkItems = chunk(workspaces, 5);
+
+    for (const chunkItem of chunkItems) {
+      await Promise.all(
+        chunkItem.map(async (workspace) => {
```

Comment:
If we have 1200 active workspace this would probably block the API and break the domain setup feature for everyone https://developers.cloudflare.com/fundamentals/api/reference/limits/

We could maybe instead use cloudflare domains as source of truth we iterate on, instead of iterating on workspaces? Or at least paginate/pass multiple ID and gain one order of magnitude before this breaks 😅

Reply:
We’re still within safe limits, as I only retrieve workspaces that have a custom domain.